packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -54,12 +56,15 @@ export class H5WasmApi extends ProviderApi {
     const h5wDataset = await this.getH5WasmEntity(dataset.path);
     assertH5WasmDataset(h5wDataset);
 
-    if (hasBoolType(dataset)) {
-      const value = h5wDataset.to_array();
+    // h5wasm returns integers for bool and BigInt for int64
+    // So we use to_array instead to have bool and numbers resp.
+    if (hasBoolType(dataset) || hasInt64Type(dataset)) {
+      const rawValue = h5wDataset.to_array();
+      const value = hasArrayShape(dataset)
+        ? flattenValue(rawValue, dataset)
+        : rawValue;
```

Comment:
2. As `to_array` returns nested arrays for nD datasets and as the visus expect 1D array, flatten it.

The previous code was flattening the result of `to_array` only if there was slicing. That means that 2D arrays were not flattened for 2D+ vis (`Matrix`, `Heatmap`),  which was breaking them.